### PR TITLE
Add Topic Statistics Example

### PR DESCRIPTION
**Pull request by dabonnie**
Adds an example subscriber which enables topic statistics and illustrates some of the configuration options.

Related to https://github.com/ros2/ros2_documentation/pull/824, https://github.com/ros-tooling/aws-roadmap/issues/246

Signed-off-by: Devin Bonnie <dbbonnie@amazon.com>

### CODE REVIEW

**Comment by dabonnie** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11542/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6705/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9440/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/11475/)

**Comment by dabonnie** [comment]
PTAL @jacobperron @hidmic

**Comment by dabonnie** [comment]
@jacobperron @hidmic friendly ping :-)

File: rclcpp/topics/minimal_subscriber/member_function_with_topic_statistics.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+/* This example creates a subclass of Node and uses std::bind() to register a
+ * member function as a callback from the timer. */
+
+class MinimalPublisher : public rclcpp::Node
+{
+public:
+  MinimalPublisher()
+  : Node("minimal_publisher"), count_(0)
+  {
+    publisher_ = this->create_publisher<std_msgs::msg::String>("topic", 10);
```
Comment (jacobperron) [inline rclcpp/topics/minimal_subscriber/member_function_with_topic_statistics.cpp:32]: I'm confused. This looks like a regular publisher example, I guess you meant to upload a subscriber example (with topic stats enabled) instead.

Comment (dabonnie) [reply rclcpp/topics/minimal_subscriber/member_function_with_topic_statistics.cpp:32]: Wow, I'm confused too. Sorry and my mistake: let me figure out where I went wrong.

Comment (dabonnie) [reply rclcpp/topics/minimal_subscriber/member_function_with_topic_statistics.cpp]: Ok, still not sure how it happened but the correct contents are now in the PR.

**Comment by dabonnie** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11589/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6730/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9468/)
* Windows [Build Status]

**Pull request merged**